File: MultiLanguageSandbox/src/test/deps/ruby/46.rb


# Counts the number of characters in a given title, excluding spaces and newline characters.
# The function takes a single argument, title, which is a string representing the title of an essay.
# It returns the count of characters in the title, excluding any spaces and newline characters.
#
# Examples:
# >>> count_chars_in_title("234")
# 3
# >>> count_chars_in_title("Ca 45")
# 4
# >>> count_chars_in_title("Hello World\n")
# 10

def count_chars_in_title(title)
  # Remove all spaces and newline characters from the title
  cleaned_title = title.gsub(/[ \n]/, '')
  # Return the length of the cleaned string
  cleaned_title.length
end

raise 'Test failed' unless count_chars_in_title("234") == 3
raise 'Test failed' unless count_chars_in_title("Ca 45") == 4
raise 'Test failed' unless count_chars_in_title("Hello World\n") == 10
raise 'Test failed' unless count_chars_in_title("Ruby Programming 101 ") == 18
raise 'Test failed' unless count_chars_in_title("\nNew Line\n") == 7

File: MultiLanguageSandbox/src/test/deps/julia/46.jl
function count_composition_title_chars(title::String) :: Int

    """
    Count the number of characters in a composition title, excluding spaces and newline characters.
    
    # Examples
    - `count_composition_title_chars("234")` returns `3`.
    - `count_composition_title_chars("Ca 45")` returns `4`.
    - `count_composition_title_chars("Hello\nWorld")` returns `10`.
    - `count_composition_title_chars(" A B C ")` returns `3`.
    """
    # Filter out spaces and newline characters, then count the remaining characters
    filtered_chars = filter(c -> c != ' ' && c != '\n', title)
    return length(filtered_chars)
end
@assert count_composition_title_chars("234") == 3
@assert count_composition_title_chars("Ca 45") == 4
@assert count_composition_title_chars("Hello\nWorld") == 10
@assert count_composition_title_chars(" A B C ") == 3
@assert count_composition_title_chars("\n\n") == 0